httpx/errors/not_found.go

```diff
@@ -0,0 +1,15 @@
+package errors
+
+func NotFoundError() *notFound {
+	return &notFound{}
+}
+
+type notFound struct{}
+
+func (n *notFound) Error() string {
+	return "Not found"
+}
+
+func (n *notFound) StatusCode() int {
+	return 404
+}
```

Comment:
I think this is fine. However, since there is no input (like an error object to wrap), you _could_ expose a variable instead.

``` golang

type notFound struct{}

func (n notFound) Error() string {
	return "Not found"
}

func (n notFound) StatusCode() int {
	return 404
}

var NotFoundError = notFound{}
```